flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -254,6 +262,58 @@ public void failJobIfNotEnoughResources() throws Exception {
 				"Could not allocate the required slot within slot request timeout.").isPresent());
 	}
 
+	@Test
+	public void vertexIsResetBeforeRestarted() throws Exception {
+		final JobGraph jobGraph = singleNonParallelJobVertexJobGraph();
+
+		final TestSchedulingStrategy.Factory schedulingStrategyFactory = new TestSchedulingStrategy.Factory();
+		final DefaultScheduler scheduler = createScheduler(jobGraph, schedulingStrategyFactory);
+		final TestSchedulingStrategy schedulingStrategy = schedulingStrategyFactory.getLastCreatedSchedulingStrategy();
+		final SchedulingTopology topology = schedulingStrategy.getSchedulingTopology();
+
+		startScheduling(scheduler);
+
+		final SchedulingExecutionVertex vertex = Iterables.getOnlyElement(topology.getVertices());
+		final ExecutionAttemptID attemptId = scheduler.getExecutionVertex(vertex.getId()).getCurrentExecutionAttempt().getAttemptId();
+
+		schedulingStrategy.schedule(Collections.singleton(vertex.getId()));
+
+		scheduler.updateTaskExecutionState(new TaskExecutionState(jobGraph.getJobID(), attemptId, ExecutionState.FAILED));
+
+		taskRestartExecutor.triggerScheduledTasks();
+
+		Set<ExecutionVertexID> verticesToRestart = schedulingStrategy.getReceivedVerticesToRestart();
+		assertEquals(1, verticesToRestart.size());
+		assertEquals(ExecutionState.CREATED, topology.getVertexOrThrow(verticesToRestart.iterator().next()).getState());
```

Comment:
What is the difference to `assertEquals(ExecutionState.CREATED, vertex.getState())`? `SchedulingExecutionVertex` always reflects the newest state (it's not a snapshot).

Also see my other comment about `ArchivedExecutionVertex`.

Reply:
You are right. We can use `vertex.getState()` directly given that we know the test topology well.

Reply:
done.